packages/utils/src/instrument.ts

```diff
@@ -221,47 +290,59 @@ function instrumentXHR(): void {
       // if Sentry key appears in URL, don't capture it as a request
       // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
       if (isString(url) && xhrInfo.method === 'POST' && url.match(/sentry_key/)) {
-        xhr.__sentry_own_request__ = true;
+        this.__sentry_own_request__ = true;
       }
 
-      const onreadystatechangeHandler = function (): void {
-        if (xhr.readyState === 4) {
+      const onreadystatechangeHandler: () => void = () => {
+        // For whatever reason, this is not the same instance here as from the outer method
```

Comment:
This is extremely weird and I wasted a lot of time figuring out why this is happening, to no avail. 

basically, it seems that when we just assign to `xhrInfo` from the outer scope, this is not the same as this.__sentry_xhr__, even though the `this` remains the same, and I don't see any other place where we'd set this 😬